tests/h/models/user_identity_test.py

```diff
@@ -76,18 +93,76 @@ def test_two_cant_have_the_same_provider_and_provider_id(self, db_session):
         ):
             db_session.flush()
 
-    def test_two_can_have_the_same_provider_id_but_different_providers(
-        self, db_session
+    def test_one_user_can_have_the_same_provider_id_from_different_providers(
+        self, db_session, user
     ):
         db_session.add_all(
             [
-                models.UserIdentity(provider="provider_1", provider_unique_id="id"),
-                models.UserIdentity(provider="provider_2", provider_unique_id="id"),
+                models.UserIdentity(
+                    provider="provider_1", provider_unique_id="id", user=user
+                ),
+                models.UserIdentity(
+                    provider="provider_2", provider_unique_id="id", user=user
+                ),
             ]
         )
 
         db_session.flush()
 
+    def test_different_users_can_have_the_same_provider_id_from_different_providers(
+        self, db_session, factories
+    ):
+        db_session.add_all(
+            [
+                models.UserIdentity(
+                    provider="provider_1",
+                    provider_unique_id="id",
+                    user=factories.User(),
+                ),
+                models.UserIdentity(
+                    provider="provider_2",
+                    provider_unique_id="id",
+                    user=factories.User(),
+                ),
+            ]
+        )
+
+        db_session.flush()
+
+    def test_removing_a_user_identity_from_a_user_deletes_it(self, db_session, user):
```

Comment:
Test name is a bit ambiguous (what is `it`, the user or the identity?). Then again, other options get pretty lengthy.

Reply:
Fixed!